src/AssemblyLinter.js

```diff
@@ -63,34 +63,32 @@ class AssemblyLinter {
 
         const diagnostics = [];
         const lines = document.getText().split('\n');
+        const text = document.getText(); // for multi-line rules
     
         // Load the linting rules from the JSON file
         const rulesPath = path.join(__dirname, 'rules.json');
         const rules = JSON.parse(fs.readFileSync(rulesPath, 'utf8')).rules;
     
         for (const rule of rules) {
-            const regex = new RegExp(rule.pattern, 'g');
+            const regex = new RegExp(rule.pattern, 'gd');
```

Comment:
This was clutch for getting the correct index for the error locations, thank you @lettucegoblin !